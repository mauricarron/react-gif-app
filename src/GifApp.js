import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <div className="search-bar">
        <h1>.gifAPP</h1>
        <AddCategory setCategories={setCategories} />
      </div>
      <div>
        {categories.map((category, i) => (
          <GifGrid category={category} key={`${i}-${category}`} />
        ))}
      </div>
    </>
  );
};

export default GifApp;
