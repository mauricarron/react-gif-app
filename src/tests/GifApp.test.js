import { shallow } from "enzyme";
import GifApp from "../GifApp";

describe("tests on <GifApp />", () => {
  const wrapper = shallow(<GifApp />);

  test("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should match components with categories", () => {
    const categories = ["Kimetsu no yaiba", "jujutsu kaisen"];

    const wrapper = shallow(<GifApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
