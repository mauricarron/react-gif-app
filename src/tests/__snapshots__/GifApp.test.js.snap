// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifApp /> should match components with categories 1`] = `
<Fragment>
  <div
    className="search-bar"
  >
    <h1>
      .gifAPP
    </h1>
    <AddCategory
      setCategories={[Function]}
    />
  </div>
  <div>
    <GifGrid
      category="Kimetsu no yaiba"
      key="0-Kimetsu no yaiba"
    />
    <GifGrid
      category="jujutsu kaisen"
      key="1-jujutsu kaisen"
    />
  </div>
</Fragment>
`;

exports[`tests on <GifApp /> should match snapshot 1`] = `
<Fragment>
  <div
    className="search-bar"
  >
    <h1>
      .gifAPP
    </h1>
    <AddCategory
      setCategories={[Function]}
    />
  </div>
  <div />
</Fragment>
`;
